packages/blockchain-extension/extension/explorer/walletExplorer.ts

```diff
@@ -92,48 +98,70 @@ export class BlockchainWalletExplorerProvider implements BlockchainExplorerProvi
                 walletGroups[groupIndex].push(walletRegistryEntry);
             } else {
                 // Create new group
-                walletGroups.push([walletRegistryEntry]);
+                if (walletRegistryEntry.fromEnvironment) {
+                    walletGroups.push([walletRegistryEntry]);
+                } else {
+                    // group wallets that don't belong to environments
+                    // don't add wallet if it doesn't have a wallet path
+                    if (walletRegistryEntry.walletPath) {
+                        otherWallets.push(walletRegistryEntry);
+                    }
+                }
             }
         }
+        if (otherWallets.length > 0) {
+            walletGroups.push(otherWallets);
+        }
 
         if (walletGroups.length === 0) {
             tree.push(new TextTreeItem(this, 'No wallets found'));
         } else {
 
             for (const group of walletGroups) {
-                if (group.length === 1) {
-                    const walletTreeItems: WalletTreeItem[] = await this.populateWallets(group); // There will only be one
-                    if (walletTreeItems.length > 0) {
-                        // Only push if there's a wallet path
-                        tree.push(walletTreeItems[0]);
-                    }
-                } else {
-                    // Group length is greater than 1
-                    tree.push(new WalletGroupTreeItem(this, group[0].fromEnvironment, group, vscode.TreeItemCollapsibleState.Expanded));
+                const groupName: string = group[0].fromEnvironment ? group[0].fromEnvironment : 'Other wallets';
+                const groupTreeItem: WalletGroupTreeItem = new WalletGroupTreeItem(this, groupName, group, vscode.TreeItemCollapsibleState.Expanded);
+                if (group[0].fromEnvironment) {
+                    groupTreeItem.iconPath = await this.getWalletGroupIcon(group[0]);
                 }
+                tree.push(groupTreeItem);
             }
         }
 
         return tree;
     }
 
+    private async getWalletGroupIcon(wallet: FabricWalletRegistryEntry): Promise<{ light: string | vscode.Uri; dark: string | vscode.Uri }> {
```

Comment:
This is pretty much the same code as `getGatewayGroupIcon()` - perhaps we should just make one function but take in a `FabricWalletRegistryEntry` or `FabricGatewayRegistryEntry`.
Maybe put it in a `CommonExplorer` class or something?